src/main/java/cn/nukkit/blockentity/BlockEntityHopper.java

```diff
@@ -290,43 +290,44 @@ public boolean transferItemsOut() {
                 if (inventory.isFull()) {
                     return false;
                 }
-                for (int i = 0; i < inventory.getSize(); i++) {
+
+                for (int i = 0; i < this.inventory.getSize(); i++){
                     Item item = this.inventory.getItem(i);
-                    if (item.getId() != 0 && item.getCount() > 0) {
+                    if (item.getId() != 0 && item.getCount() > 0){
                         Item itemToAdd = item.clone();
                         itemToAdd.setCount(1);
-                        //If it is a fuel
-                        if (Fuel.duration.containsKey(item.getId())) {
-                            if (inventory.getFuel().getId() == Item.AIR) {
-                                inventory.setFuel(itemToAdd);
-                                item.count--;
-                            } else if (inventory.getFuel().getId() == itemToAdd.getId()) {
-                                Item fuel = inventory.getFuel();
-                                if (fuel.count < fuel.getMaxStackSize())
-                                {
-                                    fuel.count++;
-                                    inventory.setFuel(fuel);
-                                    item.count--;
-                                }
-                            }
-                        } else { //Must be an input, try to cook it
-                            if (inventory.getSmelting().getId() == Item.AIR) {
+                        
+                        //Check direction of hopper
+                        if (this.getBlock().getDamage() == 0){
+                            if (inventory.getSmelting().getId() == Item.AIR){
                                 inventory.setSmelting(itemToAdd);
                                 item.count--;
-                            } else if (inventory.getSmelting().getId() == itemToAdd.getId()) {
```

Comment:
Indentation is incorrect. There should be one space before the curly bracket.

Reply:
I just now fixed this in all of the parts I edited.